File: qc-storage/src/main/java/com/eager/questioncloud/storage/question/converter/PostFileConverter.java
package com.eager.questioncloud.storage.question.converter;

import com.eager.questioncloud.core.domain.hub.board.vo.PostFile;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import jakarta.persistence.AttributeConverter;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class PostFileConverter implements AttributeConverter<List<PostFile>, String> {
    private final ObjectMapper objectMapper = new ObjectMapper();

    @Override
    public String convertToDatabaseColumn(List<PostFile> attribute) {
        if (attribute == null || attribute.isEmpty()) {
            return null;
        }
        try {
            return objectMapper.writeValueAsString(attribute);
        } catch (JsonProcessingException e) {
            return null;
        }
    }

    @Override
    public List<PostFile> convertToEntityAttribute(String dbData) {
        List<PostFile> list = new ArrayList<>();
        if (dbData == null) {
            return list;
        }
        try {
            return Arrays.asList(objectMapper.readValue(dbData, PostFile[].class));
        } catch (JsonProcessingException e) {
            return list;
        }
    }
}

File: qc-domain/qc-post/qc-post-core/src/main/kotlin/com/eager/questioncloud/post/infrastructure/converter/PostFileConverter.kt
package com.eager.questioncloud.post.infrastructure.converter

import com.eager.questioncloud.post.domain.PostFile
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.module.kotlin.registerKotlinModule
import jakarta.persistence.AttributeConverter
import java.util.*

class PostFileConverter : AttributeConverter<List<PostFile?>?, String> {
    private val objectMapper = ObjectMapper().registerKotlinModule()

    override fun convertToDatabaseColumn(attribute: List<PostFile?>?): String? {
        if (attribute.isNullOrEmpty()) {
            return null
        }

        return runCatching { objectMapper.writeValueAsString(attribute) }.getOrNull()
    }

    override fun convertToEntityAttribute(dbData: String?): List<PostFile> {
        if (dbData.isNullOrEmpty()) {
            return emptyList()
        }

        return runCatching {
            objectMapper.readValue(dbData, Array<PostFile>::class.java).toList()
        }.getOrDefault(emptyList())
    }
}
